system_tests/espresso/espresso_transaction_streamer_throughput_test.go

```diff
@@ -30,93 +108,237 @@ func TestTransactionStreamerEspressoThroughput(t *testing.T) {
+						return chain.NewSiphonBlocksWithTransactions(espressoClient, blocksWithTransactionsCh)
+					},
+				),
+				transaction_streamer.WithMultipleMockTransactionStreamerEnvironmentOptions(
+					scenario.streamerOptions...,
+				),
+			)
 
-	// Stop running the TransactionStreamer
-	runCancel()
-	streamer.StopWaiter.StopAndWait()
+			if have, want := err, error(nil); !errors.Is(have, want) {
+				t.Fatalf("encountered error while creating mock transaction streamer environment:\nhave:\n\t\"%v\"\nwant:\n\t\"%v\"", have, want)
+			}
 
-	timingData := chain.Timing(start, end)
+			// Run the scenario with the mock environment
+			RunChains(scenarioCtx, t, mockEspressoChain, streamer, scenario.behaviors...)
 
-	fmt.Printf("Polling Interval %s, Submission Interval %s => Took %s, Throughput: %.2f messages/s\n",
-		pollingInterval,
-		submissionInterval,
-		timingData.Duration,
-		float64(N)/timingData.Duration.Seconds(),
-	)
+			start := time.Now()
+			// Start the TransactionStreamer, so that processing begins
+			if have, want := streamer.Start(scenarioCtx), error(nil); !errors.Is(have, want) {
+				t.Fatalf("encountered error while starting TransactionStreamer:\nhave:\n\t\"%v\"\nwant:\n\t\"%v\"", have, want)
+			}
+
+			// Let's grab the messages that are being sent to the TransactionStreamer
+			// and are being processed by the Espresso chain.
+			receivedMessages := make(map[common.Hash]arbostypes.MessageWithMetadata)
+
+			// Let's consume the transactions from the mock espresso chain, until we get all
+			// of the transactions that we sent to the TransactionStreamer.
+			for len(receivedMessages) < scenario.samples {
+				// Read the next transaction
+				blockWithTx := <-blocksWithTransactionsCh
+
+				messages, err := generate.ConvertEspressoTransactionsInBlockToMessages(blockWithTx)
+				if have, want := err, error(nil); !errors.Is(have, want) {
+					t.Fatalf("encountered error while converting transactions in block to messages:\nhave:\n\t\"%v\"\nwant:\n\t\"%v\"", have, want)
+				}
+
+				for _, m := range messages {
+					receivedMessages[hasher.HashMessageWithMetadata(&m)] = m
+				}
+			}
+			end := time.Now()
+
+			// Stop running the TransactionStreamer
+			scenarioCancel()
+			streamer.StopWaiter.StopAndWait()
+
+			timingData := chain.Timing(start, end)
+
+			// Compute the total bytes processed
+			var totalBytesProcessed uint64
+			for _, m := range receivedMessages {
+				totalBytesProcessed += uint64(len(m.Message.L2msg))
+			}
+
+			if have, want := timingData.Duration, scenario.duration; !want.IsLessThanOrEqualTo(have) {
+				t.Errorf("duration is not within expected range:\nhave:\n\t%s\nwant:\n\t<=%s\n",
+					have,
+					want,
+				)
+			}
+
+			if have, want := float64(len(receivedMessages))/timingData.Duration.Seconds(), scenario.messageThroughput; !want.IsGreaterThanOrEqualTo(have) {
+				t.Errorf("message throughput is not within expected range:\nhave:\n\t%.2f messages/s\nwant:\n\t>= %s messages/s\n",
```

Comment:
should we also use FatalF instead of ErrorF here similar to all other checks in the file?

Reply:
The difference between `Errorf` and `Fatalf` is that an `Errorf` allows the test to continue whereas `Fatalf` stops the rest of the test from running.

If we utilized `Fatalf` we would stop after the first failure.  By utilizing `Errorf` it allows multiple failure results to be identified and seen within the same test at once.  This allows for the identification of potentially multiple failures in a single test run.  This can allow for faster iteration as you can resolve potentially multiple issues in one test failure rather than having to fix and re-run the tests one at a time.